### PR TITLE
Collect SCCs during e2e

### PR DESCRIPTION
**Pull request by enj**
Signed-off-by: Monis Khan <mkhan@redhat.com>

### CODE REVIEW

File: ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml [hunk 1]
```diff
@@ -378,10 +378,12 @@ objects:
           oc --insecure-skip-tls-verify --request-timeout=5s get pods --all-namespaces --template '{{ range .items }}{{ $name := .metadata.name }}{{ $ns := .metadata.namespace }}{{ range .spec.containers }}-n {{ $ns }} {{ $name }} -c {{ .name }}{{ "\n" }}{{ end }}{{ range .spec.initContainers }}-n {{ $ns }} {{ $name }} -c {{ .name }}{{ "\n" }}{{ end }}{{ end }}' > /tmp/containers
           oc --insecure-skip-tls-verify --request-timeout=5s get pods -l openshift.io/component=api --all-namespaces --template '{{ range .items }}-n {{ .metadata.namespace }} {{ .metadata.name }}{{ "\n" }}{{ end }}' > /tmp/pods-api
 
+          queue /tmp/artifacts/config-resources.json oc --insecure-skip-tls-verify --request-timeout=5s get apiserver.config.openshift.io authentication.config.openshift.io build.config.openshift.io console.config.openshift.io dns.config.openshift.io featuregate.config.openshift.io image.config.openshift.io infrastructure.config.openshift.io ingress.config.openshift.io network.config.openshift.io oauth.config.openshift.io project.config.openshift.io scheduler.config.openshift.io -o json
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml]: fragile.  Please use jordan's excellent hack to get the list of resource types in the group

File: ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml [hunk 2]
```diff
@@ -378,26 +378,30 @@ objects:
...
 
+          for group in config.openshift.io operator.openshift.io; do
+              queue /tmp/artifacts/"$(echo ${group} | cut -f 1 -d .)"-resources.json oc --insecure-skip-tls-verify --request-timeout=5s get $(oc api-resources --api-group ${group} -o name | tr '\n' ',' | sed 's/.$//') -o json;
+          done
           queue /tmp/artifacts/apiservices.json oc --insecure-skip-tls-verify --request-timeout=5s get apiservices -o json
-          queue /tmp/artifacts/clusteroperators.json oc --insecure-skip-tls-verify --request-timeout=5s get clusteroperators -o json
-          queue /tmp/artifacts/clusterversion.json oc --insecure-skip-tls-verify --request-timeout=5s get clusterversion -o json
           queue /tmp/artifacts/configmaps.json oc --insecure-skip-tls-verify --request-timeout=5s get configmaps --all-namespaces -o json
+          queue /tmp/artifacts/credentialsrequests.json oc --insecure-skip-tls-verify --request-timeout=5s get credentialsrequests --all-namespaces -o json
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml]: Does this contain secrets?

File: ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml [hunk 3]
```diff
@@ -378,26 +378,30 @@ objects:
...
-          queue /tmp/artifacts/kubeapiserver.json oc --insecure-skip-tls-verify --request-timeout=5s get kubeapiserver -o json
-          queue /tmp/artifacts/kubecontrollermanager.json oc --insecure-skip-tls-verify --request-timeout=5s get kubecontrollermanager -o json
           queue /tmp/artifacts/machineconfigpools.json oc --insecure-skip-tls-verify --request-timeout=5s get machineconfigpools -o json
           queue /tmp/artifacts/machineconfigs.json oc --insecure-skip-tls-verify --request-timeout=5s get machineconfigs -o json
           queue /tmp/artifacts/namespaces.json oc --insecure-skip-tls-verify --request-timeout=5s get namespaces -o json
           queue /tmp/artifacts/nodes.json oc --insecure-skip-tls-verify --request-timeout=5s get nodes -o json
-          queue /tmp/artifacts/openshiftapiserver.json oc --insecure-skip-tls-verify --request-timeout=5s get openshiftapiserver -o json
           queue /tmp/artifacts/pods.json oc --insecure-skip-tls-verify --request-timeout=5s get pods --all-namespaces -o json
+          queue /tmp/artifacts/persistentvolumes.json oc --insecure-skip-tls-verify --request-timeout=5s get persistentvolumes --all-namespaces -o json
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml]: Can these contain secrets?

File: ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml [hunk 4]
```diff
@@ -378,26 +378,30 @@ objects:
...
+          queue /tmp/artifacts/persistentvolumeclaims.json oc --insecure-skip-tls-verify --request-timeout=5s get persistentvolumeclaims --all-namespaces -o json
           FILTER=gzip queue /tmp/artifacts/replicasets.json.gz oc --insecure-skip-tls-verify --request-timeout=5s get replicasets --all-namespaces -o json
           queue /tmp/artifacts/rolebindings.json oc --insecure-skip-tls-verify --request-timeout=5s get rolebindings --all-namespaces -o json
           queue /tmp/artifacts/roles.json oc --insecure-skip-tls-verify --request-timeout=5s get roles --all-namespaces -o json
+          queue /tmp/artifacts/clusterrolebindings.json oc --insecure-skip-tls-verify --request-timeout=5s get clusterrolebindings --all-namespaces -o json
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml]: What are these for?

Comment (enj) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml]: So I can tell what all of the RBAC data of the cluster is ... ?

File: ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml (hunk 2 above)
Comment (enj) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml]: Only secret refs from what I see: https://github.com/openshift/cloud-credential-operator/blob/master/pkg/apis/cloudcredential/v1/credentialsrequest_types.go

File: ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml (hunk 3 above)
Comment (enj) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml]: I only see various secret refs in the source types.

**Comment by smarterclayton** [comment]
/hold

want to see tests pass

**Comment by enj** [comment]
/retest
/hold cancel

I can see from the manifests in the CI runs that the new files are being created with the desired content.

**Comment by enj** [comment]
@mfojtik @sttts @stlaz you may care one day.

/close

**Pull request closed without merging**